### PR TITLE
feat: support JSON Batch messages from kinesis

### PR DESCRIPTION
**Pull request by paulswartz**
We convert them into multiple lines, which are then combined normally by the packagers.

#### Summary of changes

**Asana Ticket:** [Trike reliability updates](https://app.asana.com/0/584764604969369/1203329880050199)

### CODE REVIEW

File: src/processor.ts [hunk 1]
```diff
@@ -17,12 +17,16 @@ export const handler: Handler = Sentry.wrapHandler(async ({ records }) => ({
 
 const transformRecord = ({ recordId, data }: EventRecord): ResultRecord => {
   try {
-    const event = JSON.parse(Buffer.from(data, "base64").toString());
-    const { time, data: { raw } } = struct(event, OCSEvent); // prettier-ignore
+    const events = wrapList(JSON.parse(Buffer.from(data, "base64").toString()));
+    const {time} = struct(events[0], OCSEvent); // prettier-ignore
     const datetime = localFromISO(time);
```
Comment (bklebe) [inline src/processor.ts:22]: I know this isn't part of this review but is this function DST safe? It doesn't look like it.

**Review by bklebe — COMMENTED** [review]
Just a quick question about localFromISO, we can make a ticket to fix it later since the next DST isn't for a while.

File: src/processor.ts (hunk 1 above)
Comment (paulswartz) [reply src/processor.ts:22]: I'll add some tests to be sure, but this is going from a UTC datetime to a local time, which should be safe.

Comment (bklebe) [reply src/processor.ts:22]: oh I see, sorry.

Comment (paulswartz) [reply src/processor.ts:22]: Better safe than sorry!

**Comment by paulswartz** [comment]
Deployed to dev; will check back tomorrow to make sure that the process was successful overnight and then deploy to prod.

**Pull request merged**